core/src/main/scala/org/apache/spark/sql/collection/Utils.scala

```diff
@@ -575,6 +575,39 @@ object Utils {
     SQLExecution.withNewExecutionId(ctx, queryExecution)(body)
   }
 
+  def immutableMap[A, B](m: mutable.Map[A, B]): Map[A, B] = new Map[A, B] {
```

Comment:
Not clear on question. The '+' and '-' are abstract methods in immutable.Map that need to be implemented. In mutable map these will change the original map itself but want to avoid for this, hence this wrapper (like Java's Collections.unmodifiable...)

The conversion of mutable to immutable using toMap creates a copy which I wanted to avoid since it can be invoked by many modules in future.